runtime/src/bank.rs

```diff
@@ -3086,6 +3821,111 @@ impl Bank {
+    /// Process reward credits for a partition of rewards
+    ///     - partition_index: reward partition index
+    /// Store the rewards to AccountsDB, update reward history record and total capital.
+    fn credit_epoch_rewards_in_partition(&mut self, partition_index: u64) {
+        if let EpochRewardStatus::Active(StartBlockHeightAndRewards {
+            parent_start_block_height: _parent_start_block_height,
+            calculated_epoch_stake_rewards: ref stake_rewards,
+        }) = self.epoch_reward_status
+        {
+            let mut metrics = RewardsStoreMetrics {
+                pre_capitalization: self.capitalization(),
+                total_stake_accounts_count: stake_rewards.len(),
+                partition_index,
+                ..RewardsStoreMetrics::default()
+            };
+
+            let this_partition_stake_rewards =
+                &stake_rewards[self.get_partition_range(partition_index, stake_rewards.len())];
+
+            let (
+                DistributedRewardsSum {
+                    num_rewards: stake_store_counts,
+                    total_rewards_in_lamports: total_stake_rewards,
+                },
+                measure_us,
+            ) = measure_us!(self.store_stake_accounts_in_partition(this_partition_stake_rewards));
+            metrics.store_stake_accounts_us += measure_us;
+            metrics.store_stake_accounts_count += stake_store_counts;
+
+            self.update_reward_history_in_partition(this_partition_stake_rewards);
+
+            // increase total capital by the distributed rewards
+            self.capitalization
+                .fetch_add(total_stake_rewards as u64, Relaxed);
+
+            // update EpochRewards sysvar with distributed rewards (decrease total capital by distributed rewards)
+            self.update_epoch_rewards(total_stake_rewards as u64);
+
+            metrics.post_capitalization = self.capitalization();
+
+            report_partitioned_reward_metrics(self, metrics);
+        }
+    }
+
+    /// Create EpochRewards syavar with calculated rewards
+    fn create_epoch_rewards(
+        &self,
+        total_rewards: u64,
+        distributed_rewards: u64,
+        distribution_complete_block_height: u64,
+    ) {
+        assert!(self.partitioned_rewards_feature_enabled());
+
+        let epoch_rewards = sysvar::epoch_rewards::EpochRewards::new(
+            total_rewards,
+            distributed_rewards,
+            distribution_complete_block_height,
+        );
+
+        self.update_sysvar_account(&sysvar::epoch_rewards::id(), |account| {
+            let mut inherited_account_fields =
+                self.inherit_specially_retained_account_fields(account);
+
+            assert!(total_rewards >= distributed_rewards);
+            // set the account lamports to the undistributed rewards
+            inherited_account_fields.0 = total_rewards - distributed_rewards;
+            create_account(&epoch_rewards, inherited_account_fields)
+        });
```

Comment:
It looks like the sysvar account deletion code doesn't exist yet, but since `update_sysvar_account` will create enough lamports for the account to be rent-exempt, be sure to burn those lamports after the final distribution to delete the account

Reply:
yeah. see bank.rs:1932.